pkg/ccl/backupccl/restore_job.go

```diff
@@ -1512,6 +1520,64 @@ func createImportingDescriptors(
 	return dataToPreRestore, dataToRestore, nil
 }
 
+// remapPublicSchemas is used to create a descriptor backed public schema
+// for databases that have virtual public schemas.
+// The rewrite map is updated with the new public schema id.
+func remapPublicSchemas(
+	ctx context.Context,
+	p sql.JobExecContext,
+	mutableDatabases []*dbdesc.Mutable,
+	schemasToWrite *[]*schemadesc.Mutable,
+	writtenSchemas *[]catalog.SchemaDescriptor,
+	details *jobspb.RestoreDetails,
+) error {
+	databaseToPublicSchemaID := make(map[descpb.ID]descpb.ID)
+	// Create descriptor backed Public schema if necessary.
+	for _, db := range mutableDatabases {
+		if db.Schemas == nil || !db.HasPublicSchemaWithDescriptor() {
+			id, err := catalogkv.GenerateUniqueDescID(ctx, p.ExecCfg().DB, p.ExecCfg().Codec)
+			if err != nil {
+				return err
+			}
+
+			db.Schemas = map[string]descpb.DatabaseDescriptor_SchemaInfo{
```

Comment:
We are overwriting any other Schemas that might have been already written in `rewriteDatabaseDescs`. I don't think this is correct.